include/simulation_time/simulation_clock.hpp

```diff
@@ -0,0 +1,205 @@
+//! calendar time.
+//!
+//! Implements named requirements: Clock (https://en.cppreference.com/w/cpp/named_req/Clock)
+//!
+struct simulation_clock final
+{
+    /* Clock named requirements */
+
+    using rep = long; // TODO: double vs std::int64_t
+    using period = std::ratio<1>;
+    using duration = std::chrono::duration<rep, period>;
+    using time_point = std::chrono::time_point<simulation_clock, duration>;
+    static constexpr bool is_steady = false;
+
+    //! Current Simulation Time
+    //! \returns The current point in time the simulation is at.
+    //! \post Return value must have a non-negative representation value..
+    static time_point now() noexcept;
+
+    /* simulation_clock-specific types/interface */
+
+    // explicit simulation time type aliases //
+    using simulation_rep = rep;
+    using simulation_period = period;
+    using simulation_duration = duration;
+    using simulation_time_point = time_point;
+
+    // explicit calendar time type aliases //
+    using calendar_clock = std::chrono::system_clock;
+    using calendar_rep = calendar_clock::rep;
+    using calendar_period = calendar_clock::period;
+    using calendar_duration = calendar_clock::duration;
+    using calendar_time_point = calendar_clock::time_point;
+
+    // simulation-calendar conversion functions //
+
+    //! Convert a calendar time point to a simulation time point.
+    //! \param calendar_time Calendar time point.
+    //! \returns The corresponding simulation time point.
+    //! \pre simulation_clock::set_epoch must have been called prior to calling this.
+    static simulation_time_point from_calendar(const calendar_time_point& calendar_time);
+
+    //! Convert a simulation time point to a calendar time point.
+    //! \param sim_time Simulation time point.
+    //! \returns The corresponding calendar time point.
+    //! \pre simulation_clock::set_epoch must have been called prior to calling this.
+    static calendar_time_point to_calendar(const simulation_time_point& sim_time);
+
+    //! Convert a std::time_t value to a simulation time point.
+    //! \see simulation_clock::from_calendar
+    //! \param time Unix epoch-based time value
+    //! \returns The corresponding simulation time point.
+    //! \pre simulation_clock::set_epoch must have been called prior to calling this.
+    static simulation_time_point from_time_t(std::time_t t);
+
+    //! Convert a simulation time point to a std::time_t value.
+    //! \see simulation_clock::to_calendar
+    //! \param sim_time Simulation time point.
+    //! \returns The corresponding std::time_t value.
+    //! \pre simulation_clock::set_epoch must have been called prior to calling this.
+    static std::time_t to_time_t(const simulation_time_point& sim_time);
+
+    // simulation management functions //
+
+    //! Unitialized epoch value, aka the epoch of typename simulation_clock::calendar_clock.
+    static constexpr auto uninitialized_epoch = calendar_time_point{};
```

Comment:
`static constexpr` isn't implicitly inline before C++17, it seems? https://en.cppreference.com/w/cpp/language/constexpr

Alternatively, this could be a `static constexpr` member function.

Reply:
Converted to a `static constexpr` member function.